### DISCUSSION TITLE
Studio Roles

### DISCUSSION THREAD

**Discussion by hrtzg** [ScratchAddons/ScratchAddons#1859] [Ideas]
Navigating the curator tab on studios provides limited information in regards to what a user's role is. A solution to this could be providing a simple syntax that could easily be parsed from a studio description and used to display roles/categories on the curator page.

For example:

```
Studio Information

[Founders: Thing1, Thing2]
[Coders: CoolCoder]
[Designers: Designz, ILoveAdobe, DesignTime]
```

Through this, if the user is listed and is a curator or manager, they can be sorted into the correct category. If no category is defined, they would keep the default role as a curator or manager. The hierarchy of roles would depend on which is defined first.

This allows individuals with Scratch Addons to not be stuck searching through a massive studio description to find out what a person's role is. For people without the extension, the syntax would still be readable and easy to understand.

**Comment by Mr-Programmer-777** [comment]
Why?

**Reply by hrtzg** [reply]
> Why?

I'd recommend reading the post. It demonstrates the usefulness of this.

**Comment by cobaltt7** [comment]
i cant see how the studio descr would be parsed

**Reply by hrtzg** [reply]
> i cant see how the studio descr would be parsed

The syntax provided allows it to be both readable and parsed. You can check to see if the line contains [:] and if the text between [here: contains no spaces. From there, you'd split everything after via the comma divider and only allow characters a-z, 0-9, and -_ through a simple regex expression.
